### PR TITLE
Add build constraint to allow compiling without `cgo`

### PR DESCRIPTION
**Pull request by alfrunes**
The PKCS11 feature (not used by the backend) prevents compiling mender-artifact without enabling `cgo`.

### CODE REVIEW

**Review by tranchitella — APPROVED** [review]

**Review by kacf — CHANGES_REQUESTED** [review]
I think this should be made into a feature flag (such as `nopkcs11` or something like that). I think silently disabling features if cgo is disabled is surprising and unintuitive.

However, independent from that, I don't think it will work as intended, because I think lzma requires cgo as well, and we can't disable that. Not 100% sure though.

**Comment by alfrunes** [comment]
> I think this should be made into a feature flag (such as `nopkcs11` or something like that). I think silently disabling features if cgo is disabled is surprising and unintuitive.

This is exactly how we do it for `lzma` (I found out after filing this PR), and I agree that disabling features should be explicit. However, this feature abuse the built-in `linux` constraint which makes impossible to opt out (in the backend we do not use this dependency). I will change the build tag to `pkcs11` and have it set by default in the Makefile.

> However, independent from that, I don't think it will work as intended, because I think lzma requires cgo as well, and we can't disable that. Not 100% sure though.

For `lzma` I will similarly add a different (Go) dependency if compiled without cgo since this one is crucial.

**Comment by alfrunes** [comment]
Thoughts on this one? #605

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Keep golang version aligned with latest yocto release
-FROM golang:1.17.13-bullseye as builder
+FROM golang:1.22.2-bullseye as builder
```
Comment (lluiscampos) [inline Dockerfile:2]: Please keep the version aligned to yocto, see comment above.

Coincidentally, we are discussing this in the [upcoming client team meeting](https://docs.google.com/document/d/1neXnS16brsb7Jdn-Sg-f1Wyoyo_O5MJIAW7_FzTKpfY/edit#heading=h.ih2nzgp6q407).

**Review by lluiscampos — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (alfrunes) [reply Dockerfile:2]: Please keep in mind that `go1.17` has been out of support for almost 2 years. Is it not possible to use the most recent version of Go when compiling Yocto?

Comment (lluiscampos) [reply Dockerfile:2]: Yocto maintains the toolchain, porting security fixes (same as for example Debian or other OSs do for the same time period that they support the OS).

Fixing the version here does have any impact in the final yocto build nor in the deb packages build.

Fixing the version here is only a way for us (me) to early catch when updates will break in today *latest* yocto LTS. Then we can decide to accept the breakage and patch the code there or reject the update for the time being. It is a difficult balance - to give you a small nitpick: we still support `dunfell` LTS which uses golang 1.14 and it is really hard and time consuming delivering our software there.

**Comment by alfrunes** [comment]
Superseded by #605

**Pull request closed without merging**